test/parallel_api/iterator/input_data_sweep.pass.cpp

```diff
@@ -0,0 +1,330 @@
+// -*- C++ -*-
+//===-- input_data_sweep.pass.cpp -----------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/utils.h"
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+
+#include "support/utils_invoke.h"
+
+// Enable for verbose list of tests run
+//#define VERBOSE_TEST 1
+
+//This test is written without indirection from invoke_on_all_hetero_policies to make clear exactly which types
+// are being tested, and to limit the number of types to be within reason.
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <int __recurse, int __reverses, bool __read = true, bool __reset_read = true, bool __write = true,
+          bool __check_write = true, bool __usable_as_perm_map = true, bool __usable_as_perm_src = true,
```

Comment:
I believe you are allowed to treat template types and values independently, and can have defaults like this without issue.  
Yes, it would perhaps be better to collapse and pack these into a single value.  It has grown quite large, I'll see what I can come up with.

Reply:
Do you have a good suggestion for cleaning this up without over-engineering or obfuscating the tests? 

It needs to be compile time, the configuration is cumulative (more an more parts of the test may get turned of with layers of recursion), and its not necessarily easy to pull apart the final wrapped iterators to get to their fundamental sequence / important wrappers to determine these values from the types themselves.

I could group them into a single struct, but that doesn't change much.  Its ugly as is, but I don't think its confusing especially with the comments.  I'm inclined to leave it as is unless there is a more specific suggestion of how to improve it.